### PR TITLE
Return native floats when decoding EclSum files

### LINKED ISSUE #2537: EclSum decoder should return native float, not numpy.float64
This line produces `numpy.float64` dictionary values:

https://github.com/equinor/ert/blob/be5cb726ff67e73a579a5c45806d6b3330e517c5/ert/serialization/_serializer.py#L96

When dumped to yaml, these are encoded as numpy types, which seems tricky to reload again using `yaml.safe_load()`. 

```
yaml.constructor.ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/object/apply:numpy.core.multiarray.scalar'
  in "<unicode string>", line 1, column 24:
    '2016-01-01 00:00:00': !!python/object/apply:numpy.core ... 
```

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #2537 

**Approach**
Applies `float()` to all elements. An alternative would be to do numpy's `.item()` which would also work if we needed to support integers, but that would not correspond to a supported `RecordType`.

### CODE REVIEW

File: ert/serialization/_serializer.py [hunk 1]
```diff
@@ -87,10 +87,14 @@ async def encode_to_path(
 
     async def decode_from_path(
         self, path: Union[str, Path], *args: Any, **kwargs: Any
-    ) -> Any:
+    ) -> Dict[str, float]:
```
Comment (xjules) [inline ert/serialization/_serializer.py:90]: Is it ok that the base class returns `Any`?

Comment (berland) [reply ert/serialization/_serializer.py:90]: No complaints from `mypy` at least on this covariant return type.

Comment (jondequinor) [reply ert/serialization/_serializer.py:90]: The logic I used was that the serialiser interface could not attempt to restrict the return type at all. A serialiser could very well return an object or a dict, right?

**Review by xjules — APPROVED** [review]
:rocket: :guitar:

**Pull request merged**